test/page/questions/questions_view_model_test.dart

```diff
@@ -33,28 +44,79 @@ void main() {
-
-      final expectedQuestions = surveyDetail.questions;
+      final questionsUiModel = await getQuestionsUiModel();
+      final expectedQuestions = questionsUiModel.questions;
       expectedQuestions.removeWhere((question) =>
           question.displayType == DisplayType.intro ||
           question.displayType == DisplayType.outro);
       expectedQuestions.sort((question1, question2) =>
           question1.displayOrder.compareTo(question2.displayOrder));
-
       final stateStream = questionsViewModel.stream;
       final questionsStream = questionsViewModel.questions;
 
-      expect(stateStream, emitsInOrder([const QuestionsState.success()]));
+      expect(stateStream, emitsInOrder([const QuestionsState.initSuccess()]));
       expect(questionsStream, emitsInOrder([expectedQuestions]));
 
-      questionsViewModel.getQuestions(surveyDetail);
+      questionsViewModel.getQuestions(questionsUiModel);
+    });
+
+    test(
+        'When calling submit survey with Success result, it returns SubmitSurveySuccess state with corresponding outroMessage',
+        () async {
+      final questionsUiModel = await getQuestionsUiModel();
+      final outroMessage = questionsUiModel.questions
+          .firstWhereOrNull(
+              (question) => question.displayType == DisplayType.outro)
+          ?.text;
+      when(mockSubmitSurveyUseCase.call(any))
+          .thenAnswer((_) async => Success(null));
+      final stateStream = questionsViewModel.stream;
+
+      expect(
+          stateStream,
+          emitsInOrder([
+            const QuestionsState.initSuccess(),
+            const QuestionsState.loading(),
+            QuestionsState.submitSurveySuccess(outroMessage),
+          ]));
+
+      questionsViewModel.getQuestions(questionsUiModel);
+      questionsViewModel.submitSurvey();
+    });
+
+    test(
+        'When calling submit survey with Failed result, it returns Error state with corresponding errorMessage',
+        () {
+      final mockException = MockUseCaseException();
+      when(mockException.actualException)
+          .thenReturn(NetworkExceptions.badRequest());
+      when(mockSubmitSurveyUseCase.call(any))
+          .thenAnswer((_) async => Failed(mockException));
+      final stateStream = questionsViewModel.stream;
+
+      expect(
+          stateStream,
+          emitsInOrder([
+            const QuestionsState.loading(),
+            QuestionsState.error(
+              NetworkExceptions.getErrorMessage(NetworkExceptions.badRequest()),
+            ),
+          ]));
+
+      questionsViewModel.submitSurvey();
     });
   });
 }
+
+Future<QuestionsUiModel> getQuestionsUiModel() async {
+  final surveyDetailJson =
+      await FileUtils.loadFile('test/mock/mock_response/survey_detail.json');
+  final surveyDetailResponse = SurveyDetailResponse.fromJson(surveyDetailJson);
+  final surveyDetail = SurveyDetailModel.fromResponse(surveyDetailResponse);
+  return QuestionsUiModel(
+    surveyId: 'surveyId',
```

Comment:
```suggestion
    surveyId: surveyDetail.surveyId,
```

surveyDetail.id = surveyDetailResponse.id